tests/torch/data/reference_graphs/quantized/synthetic_model/MultiOutputSameTensorModel.dot

```diff
@@ -2,12 +2,14 @@ strict digraph  {
 "0 /nncf_model_input_0" [id=0, type=nncf_model_input];
 "1 SymmetricQuantizer/symmetric_quantize_0" [id=1, type=symmetric_quantize];
 "2 MultiOutputSameTensorModel/__mul___0" [id=2, type=__mul__];
-"3 /nncf_model_output_0" [id=3, type=nncf_model_output];
-"4 /nncf_model_output_1" [id=4, type=nncf_model_output];
-"5 /nncf_model_output_2" [id=5, type=nncf_model_output];
+"3 /clone_0" [id=3, type=clone];
+"4 /nncf_model_output_0" [id=4, type=nncf_model_output];
+"5 /nncf_model_output_1" [id=5, type=nncf_model_output];
+"6 /nncf_model_output_2" [id=6, type=nncf_model_output];
 "0 /nncf_model_input_0" -> "1 SymmetricQuantizer/symmetric_quantize_0";
 "1 SymmetricQuantizer/symmetric_quantize_0" -> "2 MultiOutputSameTensorModel/__mul___0";
-"1 SymmetricQuantizer/symmetric_quantize_0" -> "3 /nncf_model_output_0";
-"1 SymmetricQuantizer/symmetric_quantize_0" -> "5 /nncf_model_output_2";
-"2 MultiOutputSameTensorModel/__mul___0" -> "4 /nncf_model_output_1";
+"1 SymmetricQuantizer/symmetric_quantize_0" -> "3 /clone_0";
```

Comment:
Is clone operator expected with such model?
 
 ![image](https://user-images.githubusercontent.com/4014476/139683187-64e27a89-5cda-4151-b0f4-1f39ef205997.png)
 before
 
![image](https://user-images.githubusercontent.com/4014476/139683427-8df86a80-dcaa-441e-a323-a3be640dfb23.png)

 
 now
 
![image](https://user-images.githubusercontent.com/4014476/139683391-6fe86677-eda1-4610-bfbf-e957052ed918.png)

Reply:
It happens because of our implementation of `default_dummy_forward_fn()`, behind which we use `replicate_same_tensors()`, which uses `toch.clone()`
@vshampor, please, comment

Reply:
In this PR NNCF traces the `clone` operation and adds it to the model graph, while before NNCF traced it but didn't add to the graph using ForwardTraceOnly context.

Reply:
Can we call `replicate_same_tensors` in `ForwardTraceOnly` mode to avoid adding auxiliary operations, like `clone`?

Reply:
Currently, we removed `ForwardTraceOnly`, as now we want to trace and analyze all operations. @vshampor

Reply:
The `clone` call in `replicate_same_tensor` can be done within the `with forward_nncf_trace()` context which has been introduced for these kinds of applications specifically.

Reply:
Now it is fixed. There is no more `clone` operation in the ref_graphs produced by `replicate_same_tensor`